Design-Docs/entropy.md

```diff
@@ -0,0 +1,75 @@
+---
+title: ADR SCS Entropy
+version: 2023-02-06-001
+authors: Mathias Fechner
+state: v0.9 
+---
+
+## 1. Terminology of Entropy
+
+### 1.1. Entropy in information Technology
+
+    Entrophy is a concecpt that is widely used in a scope of information 
+    technolgy. It is a measure of the amount of disorder or radomness in
+    a system. Entrophy is used to measure the amount of information in a
+    self contained systems, as well as the amount of incertitude that exists
+    in this system. For Cryptographic procedures and operation is a good 
+    entropy a must have !  
+
+    In tradional Baremetal System the amount of incertitude is or was generated by the 
+    radonmes Ready/write cycle of the disk heads of a diskdrive,  bus timeings or
+    Keyboard Timeings.
+
+### 1.2 Entropy in Virtual Instances
+
+     A virtual instance or Virtual machines has all this radomness sources not. A instance will operate normal. But if cryptographic operations will happen,
+     Producedures will take an abnormal long time, because with a small entrophy
+     count cryptographic operation can not operate in realtime. 
+     some examples are malefunctions of Applictation, OpenSSL Operation wont work. 
+
+``
+     #cat /proc/sys/kernel/random/entropy_avail 
+     256
+``
+
+#### 1.2.1 How to generate Entropy "Out-Of-Nothing" ?
+
+    One Procedure from the past in virtual machines or virtual appliances
+    have to use an entropy daemon, to ensure to get a sufficent entrophy.
+    this is today a common operation althought for embedded devices.
+    HavegeD is one example which use the combination of the Linux radom device
+
+    [haveged](http://www.issihosts.com/haveged/)
+
+``
+     #cat /proc/sys/kernel/random/entropy_avail 
+      1956
+``
+
+#### 1.2.2 CPU Hardware random number generator
+
+     Modern Server CPU's of ARM, AMD and Intel shipped Hardware random 
+     number generator. This Feature will pass-through to the virtualistaion
+     layer. This will addressed by virtio-rng.
+
+     On Baremetal-systems and virtual Instances will need the rng-tools or 
+     rng-utils, which will need to running.
+
+``
+     #cat /proc/sys/kernel/random/entropy_avail
+     3843
+``
+
+### 1.3  Entropy in SCS Clouds
+
+    in all Flavors have activated the relevant attributes:
+    
+    ```
+    hw_rng:allowed=True
+    hw_rng:rate_bytes - The allowed amount of bytes for the the guest to read from the host’s entropy per period. hw_rng:rate_period
+    ```
+    
+    in images must to activate the attribute hw_rng_model: virtio
+    at least on compute nodes must to activate the rng-utils.
+    this is requirement to guarantee to work with a confindent cryptography
+    in SCS Cloud Infrastructures.  
```

Comment:
Images must activate the[...]
On compute nodes the rng-utils must be activated.
This is a requirement to guarantee working confident cryptography[..]